### PR TITLE
Changes to ingest behavior data

### PR DESCRIPTION
**Pull request by LiuDaveLiu**
Changes to ingest behavior data

### CODE REVIEW

**Comment by wangt-github** [comment]
Some comments

In experiment.py:

1) As is, it seems that no TrialEvents (delay, sample, etc) nor ActionEvent (licks) are being populated which are necessary to reconstruct what happened.

2)

```
+            if TrialTypes[i]==0:
+                trial_instruction = 'left'
+            elif TrialTypes[i]==1:
+                trial_instruction = 'right'
```

This does not consider if reversal mode is on.

3)
```
+            StopLicking=np.where(OriginalStateNamesByNumber[i]=='StopLicking')
+            StopLicking=StopLicking[0]+1
+            Reward=np.where(OriginalStateNamesByNumber[i]=='Reward')
+            Reward=Reward[0]+1
+            TimeOut=np.where(OriginalStateNamesByNumber[i]=='TimeOut')
+            TimeOut=TimeOut[0]+1
+            NoResponse=np.where(OriginalStateNamesByNumber[i]=='NoResponse')
+            NoResponse=NoResponse[0]+1
+            EarlyLickDelay=np.where(OriginalStateNamesByNumber[i]=='EarlyLickDelay')
+            EarlyLickDelay=EarlyLickDelay[0]+1
+            EarlyLickSample=np.where(OriginalStateNamesByNumber[i]=='EarlyLickSample')
+            EarlyLickSample=EarlyLickSample[0]+1
```

It is just stylistic but these lines can be compressed.




In Ingest Behavior Data.ipynb :
```

sessionList = ['R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171114_144434.mat', 
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171116_123237.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171118_141139.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171119_122727.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171119_130846.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171120_132218.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171121_130431.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171122_134619.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171123_115915.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171124_120916.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171125_124353.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171126_122207.mat',
               'R:\Arduino\Bpod_Train1\Bpod Local\Data\dl7\TW_autoTrain\Session Data\dl7_TW_autoTrain_20171127_131041.mat']
			   
for i in range(0, len(sessionList)):
    print(i)
    session.insert1((399752, i, sessionList[i][86:90] + '-' + sessionList[i][91:93] + '-' + sessionList[i][92:94], 'daveliu',  Ingest Behavior Data.ipynb

'TRig1', sessionList[i]))
    key=session.fetch('animal', 'session')
    key1 = sessionList[i]
    session._make_tuples([key[0][i], key[1][i]], key1)
```
	

I don't think this way of adding in data will scale well:
1) When more sessions are added you can't start with i=0 for the session id. There should be some way of automatically fetching the number of sessions and auto-incrementing from that.
2) Entering data based on exact charater positions in the filename will break if the length animal name, protocol name or drive mapping changes. Considering using a regular expression.
3) Ideally instead of providing the animal ID, it can be found based on the filename.
4) Finally, it would be nice if instead of having to provide all of the filenames in an array like this, there would be some way of finding the files in a directory and only get those not already in the database.

File: lab.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import datajoint as dj
 
-schema = dj.schema('dimitri_map', locals())
+schema = dj.schema('daveliu_map', locals())
```
Comment (dimitri-yatsenko) [inline lab.py]: at some point, this should be changed to `map_lab` for everyone to access.

**Comment by ixcat** [comment]
Hi All -

Dimitri asked me to take a 1st pass review and help assist with the merging process -

Dave - glad you were able to something this going - but I think I might have gotten you started on the wrong foot with 'mimicking' the auto-ingest mechanism for this case - since the existing files modify several different tables, the 'populate()' idea gets a bit overextended and I should have checked into that possibility..

Since this is for initial ingest and eventually the data could be making it's way in via
more direct means it might make sense to take a different approach here which can also
incorporate the feedback that Tim had mentioned about file names and still keep the main table 'dj.Manual' as it is intended to be later on..

I'll come up with a quick example of what I'm getting at which I think will suit the purposes
of ingest but also keep you able to use the same schema for the data later - from there maybe we can tag up to discuss this either on slack or via some other means. Would sometime tomorrow either late morning or early afternoon work? If not other proposals welcome - definitely want you guys to be up-and running as soon as possible!

**Comment by LiuDaveLiu** [comment]
Hi,

Let's discuss 10AM Houston time.

Thank you,

Dave

**Comment by ixcat** [comment]
For anyone else - Dave & I discussed a few things and determined another ingest mechanism using a separate 'ingest' schema and table to track the files as well as a few other items - I made a skeleton in my 'dev' branch which is here:

https://github.com/ixcat/map-ephys/tree/ixcat-dev

as I understand it he's looking into combining the general idea with his code.

**Comment by LiuDaveLiu** [comment]
Closed.

**Pull request closed without merging**